### PR TITLE
Update dependencies and migrate to new API schema

### PR DESCRIPTION
**Pull request by herrherrmann**
- Updates all our npm dependencies to the latest version (except for Next.js, which has to stay on 13.4.7 due to [this bug](https://github.com/ferdikoomen/openapi-typescript-codegen/issues/1626) and [this underlying bug in SWC](https://github.com/swc-project/swc/issues/7561)).
- Adjusts our readme file and scripts to (mostly) match the https://github.com/technologiestiftung/kulturdaten-website/ repo’s setup (I will set up ESLint and Prettier in another PR).
- Migrates our API usage to the latest openAPI schema. That also means that some features like the user object in LocalStorage are currently not working, because the API is no longer returning some important fields in the response.
- Introduces `.env` with the API base URL (just like in https://github.com/technologiestiftung/kulturdaten-website/)
- Introduces an `.nvmrc` file to define which Node version to use

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1205045600493127

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -41,7 +45,5 @@ Thumbs.db
 
 notes/*
 
-+ .turbo
-
-# This folder can be generated via scripts
-src/generated-api-client/*
+# TypeScript
+*.tsbuildinfo
```
Comment (ZenVega) [inline .gitignore:49]: Here we should end with a blank line

File: README.md [hunk 2]
```diff
@@ -7,42 +7,32 @@ The kulturdaten.berlin webapp that enables cultural actors to input and manage c
...
 ## Available Scripts
 
 In the project directory, you can run:
 
-- `npm run build`: Builds the Next.js application for production usage.
+- `npm run dev`: Runs the Next.js development server.
+- `npm run generate-api`: Generates an API client from an OpenAPI specification file (`openAPI-specs.yml`).
 - `npm run format`: Runs Prettier to format source code.
+- `npm run lint`: Runs ESLint to check for linting errors.
 - `npm run run`: Runs the Next.js application with Node.js debugging enabled.
```
Comment (ZenVega) [inline README.md:32]: Still one of my (chatGPTs) bugs. But when we're here. There is one run too much.

File: README.md [hunk 3]
```diff
@@ -7,42 +7,32 @@ The kulturdaten.berlin webapp that enables cultural actors to input and manage c
...
-- "openapi-typescript-codegen": "^0.24.0"
-- "postcss": "8.4.23"
-- "react": "18.2.0"
-- "react-dom": "18.2.0"
-- "sass": "^1.62.0"
-- "tailwindcss": "3.3.1"
-- "typescript": "5.0.4"
+- `npm run typecheck`: Runs TypeScript to check for type errors.
+- `npm run build`: Builds the Next.js application for production usage.
+- `npm run start`: Starts the Next.js server for production usage.
```
Comment (ZenVega) [inline README.md:35]: actually npm start

File: src/components/LocationEditor/index.tsx [hunk 4]
```diff
@@ -18,25 +18,44 @@ const initialerrorMessages: ErrorMessages = {
...
+	const [locationObject, locationObjectSet] = useState<
+		CreateLocationRequest | Location | undefined
+	>(location || undefined);
 	const [errorMessages, errorMessagesSet] = React.useState<ErrorMessages>(initialerrorMessages);
 	const [formValid, formValidSet] = useState<boolean>(true);
 
-	const barrios = Object.entries(CreateLocation.borough).map(([key, value]) => ({
-		value: key,
-		label: value,
+	const barrios = boroughs.map((borough) => ({
```
Comment (ZenVega) [inline src/components/LocationEditor/index.tsx:51]: Here other variable names would make more sense. I think boroughs should rather be berlinDistricts and barrios could just be locationDistrics or sth like that

File: src/components/OrganisationEditor/index.tsx [hunk 5]
```diff
@@ -1,8 +1,9 @@
-import React, { useState, FC, FormEvent, useEffect } from 'react';
-import { CreateOrganization, Organization } from '../../generated-api-client';
-import { Input } from '../../components/InputField';
-import { Button } from '../../components/Button';
 import _ from 'lodash';
+import React, { FC, FormEvent, useEffect, useState } from 'react';
```
Comment (ZenVega) [inline src/components/OrganisationEditor/index.tsx:2]: Hier kann dann "React" auch raus und in Zeile 36, wo es für einen useState benutzt wird

File: src/pages/login/index.tsx [hunk 6]
```diff
@@ -44,25 +44,29 @@ const LoginPage: FC = () => {
...
 			});
+			console.log('loginResponse: ', loginResponse);
+			const loginResponseData = loginResponse.data;
+			console.log('loginResponseData: ', loginResponseData);
 			console.log('Login successful');
-			if (authObject?.authToken) {
-				setCookie('authToken', authObject.authToken, {
-					expires: authObject.expiringDate ? new Date(authObject.expiringDate) : undefined,
+			if (loginResponseData?.accessToken) {
+				setCookie('authToken', loginResponseData.accessToken, {
```
Comment (ZenVega) [inline src/pages/login/index.tsx:56]: Maybe we should also call the cookie accessToken

**Review by ZenVega — COMMENTED** [review]
Super sache!

File: README.md (hunk 3 above)
Comment (herrherrmann) [reply README.md:35]: Both works! But yeah, we should prefer `npm start`.

File: src/components/LocationEditor/index.tsx (hunk 4 above)
Comment (herrherrmann) [reply src/components/LocationEditor/index.tsx:51]: Yep! I went with `const districtOptions = berlinDistricts.map((berlinDistrict) => ...` now.

**Comment by herrherrmann** [comment]
Thanks for the feedback! I took care of all your straightforward suggestions and will merge now.

**Pull request merged**